.cspell.json

```diff
@@ -0,0 +1,147 @@
+{
+    "words": [
+        "Anush",
+        "BCSR",
+        "Bitcast",
+        "Burmako",
+        "CHLO",
+        "DCMAKE",
+        "DCSC",
+        "DCSR",
+        "DLLVM",
+        "DMLIR",
+        "Denormal",
+        "Dequantize",
+        "EBNF",
+        "ELLPACK",
+        "Einsum",
+        "FNUZ",
+        "FNUZV",
+        "Flattenings",
+        "Funcs",
+        "GEMM",
+        "GSPMD",
+        "Graphcore",
+        "IFFT",
+        "IREE",
+        "IRFFT",
+        "ITPACK",
+        "Infeed",
+        "Jakub",
+        "Lamkin",
+        "MHLO",
+        "Mehdi",
+        "Micikevicius",
+        "Nonexpanding",
+        "Noune",
+        "Numpy",
+        "Outfeed",
+        "PDLL",
+        "PHILOX",
+        "Philox",
+        "Polygamma",
+        "Proto",
+        "RFFT",
+        "Reshapable",
+        "SEEEEEMM",
+        "SEEEEMMM",
+        "SSACFG",
+        "Signless",
+        "Sunsetting",
+        "TOTALORDER",
+        "TPUs",
+        "Threefry",
+        "Unspecced",
+        "arange",
+        "ashr",
+        "backpropagating",
+        "batchnorm",
+        "bazel",
+        "bcast",
+        "bfloat",
+        "broadcastable",
+        "cbrt",
+        "ccov",
+        "classmethod",
+        "concat",
+        "cond",
+        "conv",
+        "convertable",
+        "cstr",
+        "dbgs",
+        "debuginfo",
+        "denormals",
+        "dequantization",
+        "dequantizes",
+        "dtype",
+        "ebnf",
+        "elementwise",
+        "elif",
+        "expm",
+        "floordiv",
+        "formatv",
+        "frontends",
+        "imag",
+        "infeeds",
+        "lhses",
+        "linalg",
+        "logp",
+        "lshr",
+        "matmul",
+        "memref",
+        "mlir",
+        "mmap",
+        "monoid",
+        "nonzeros",
+        "nullptr",
+        "openxla",
+        "optimise",
+        "optimisation",
+        "opset",
+        "ostream",
+        "outfeeds",
+        "parameterizing",
+        "popcnt",
+        "prettyprinted",
+        "prettyprinter",
+        "prettyprinters",
+        "prettyprinting",
+        "reinfer",
+        "requantization",
+        "rhses",
+        "roadmap",
+        "rootn",
+        "rsqrt",
+        "signfull",
+        "significand",
+        "specced",
+        "stablehlo",
+        "stddev",
+        "svarint",
+        "tablegen",
+        "testdata",
+        "umin",
+        "underflows",
+        "unflatten",
+        "unflattens",
+        "unranked",
+        "upstreaming",
+        "varint",
+        "vhlo",
+        "wellformed",
+        "workstream",
+        "workstreams",
+        "xcomplex",
+        "zext"
+    ],
+    "enableFiletypes": [
+        "cmake",
+        "tablegen"
+    ],
+    "ignoreRegExpList": [
+        "TODO\\([a-zA-Z0-9]+\\)",
+        "#ifndef [A-Z_]+",
+        "#endif .*",
+        "#define [A-Z_]+"
+    ]
+}
```

Comment:
Hmm this sounds like a considerable amount of work (and future maintenance) to avoid typos. I'm not sure the tradeoff is worth it. Can you talk more about the motivation for this change? Maybe I'm missing something.

Reply:
Happy to keep this as a local script!

Motivation is code quality, to catch spelling issues like the ones in:
- https://github.com/openxla/stablehlo/pull/1671
- https://github.com/openxla/stablehlo/pull/1674

I agree that this overhead should not be imposed on the community - that's why I don't have this set up as a CI job, just a useful script if you want to check your own files/code for spelling issues.